pkg/webhook/fleetresourcehandler/fleetresourcehandler_webhook.go

```diff
@@ -121,17 +121,28 @@ func (v *fleetResourceValidator) handleMemberCluster(req admission.Request) admi
 	return validation.ValidateUserForResource(req, v.whiteListedUsers)
 }
 
-// handleInternalMemberCluster allows/denies the request to modify internal member cluster object after validation.
-func (v *fleetResourceValidator) handleInternalMemberCluster(ctx context.Context, req admission.Request) admission.Response {
-	if req.Operation == admissionv1.Update && req.SubResource == "status" {
-		return validation.ValidateMCIdentity(ctx, v.client, req, req.Name)
+// handleFleetMemberNamespacedResource allows/denies the request to modify object after validation.
+func (v *fleetResourceValidator) handleFleetMemberNamespacedResource(ctx context.Context, req admission.Request) admission.Response {
+	var response admission.Response
+	if isFleetMemberNamespace(req.Namespace) {
+		// check to see if valid users other than member agent is making the request.
+		response = validation.ValidateUserForResource(req, v.whiteListedUsers)
+		// check to see if member agent is making the request only on Update.
+		if !response.Allowed && req.Operation == admissionv1.Update {
+			mcName := parseMemberClusterNameFromNamespace(req.Namespace)
+			if mcName != "" {
+				response = validation.ValidateMCIdentity(ctx, v.client, req, mcName)
+			}
+		}
+		return response
 	}
-	return validation.ValidateUserForResource(req, v.whiteListedUsers)
+	return admission.Allowed("namespace name doesn't begin with fleet-member prefix so we allow all operations on these namespaces")
 }
 
-// handleWork allows/delete the request to modify work object after validation.
-func (v *fleetResourceValidator) handleWork(ctx context.Context, req admission.Request) admission.Response {
-	if req.Operation == admissionv1.Update && isFleetMemberNamespace(req.Namespace) && req.SubResource == "status" {
+// handleEvent allows/denies request to modify event after validation.
+func (v *fleetResourceValidator) handleEvent(ctx context.Context, req admission.Request) admission.Response {
```

Comment:
Hi Arvind! Event is a commonly used object; if it is created outside the fleet reserved namespaces, I am afraid that it might not be a very good idea to limit its usage to whitelisted users only. -> we do have discussions on possibilities to support user controllers, e.g., custom rollout controller